src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -14,14 +16,15 @@ object UserService {
     def pure[A](a: A): DBIO[A] = DBIO.successful(a)
     def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
-
+  
   def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] = {
     val resT = for {
       validUser <- XorT[DBIO, NonEmptyList[ValidationError], User](User.validateUser(createdUser.email, createdUser.password))
-      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(validUser))
+      userId    <- XorT.right[DBIO, NonEmptyList[ValidationError], UserId](UserRepository.save(hashUserPassword(validUser)))
     } yield userId
 
     resT.value
   }
 
+  private def hashUserPassword(user: User): User = user.copy(password = user.password.bcrypt)
```

Comment:
But:

- Should the "validateUser" be responsible for hashing the password? Should it not only return the value checked if it's valid?

- If the password will be at all times in its hashed form in the DB then that should be reflected in the mapping class User? I.e. the password should be hashed by the User class whenever an instance is created?

And we would have 3 classes to basically represent the same thing, that seems like a lot of code overhead for one simple table.